clang/lib/CIR/CodeGen/CIRGenExpr.h

```diff
@@ -0,0 +1,24 @@
+//===--- CIRGenTypes.h - Type translation for CIR CodeGen -------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This is the code that handles AST -> CIR expr lowering.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_CLANG_LIB_CODEGEN_CODEGENEXPR_H
+#define LLVM_CLANG_LIB_CODEGEN_CODEGENEXPR_H
+#include "CIRGenBuilder.h"
+
+namespace cir {
+mlir::Value buildArrayAccessOp(mlir::OpBuilder &builder,
```

Comment:
Thanks for the refactoring. Since none of this depends on CIRGen, it probably makes more sense to add an implementation file for `CIRGenBuilder.cpp` and put this in `CIRGenBuilder.h` instead.